src/components/dialog/Dialog.tsx

```diff
@@ -0,0 +1,117 @@
+"use client"
+import { useSearchParams, useRouter } from 'next/navigation'
+import { useRef, useEffect, useState } from 'react';
+
+const SEARCH_URL = "http://openlibrary.org";
+
+type Props = {
+    onClose: () => void,
+    onAddToRead: () => void,
+    onAddToWantToRead: () => void,
+    loggedIn: boolean,
+}
+
+export default function Dialog({ onClose, onAddToRead, onAddToWantToRead, loggedIn }: Props) {
+  const router = useRouter()
+
+  const searchParams = useSearchParams()
+    const dialogRef = useRef<null | HTMLDialogElement>(null)
+    const showDialog = searchParams.get('showDialog')
+    const bookId = searchParams.get('search');
+
+    const [bookData, setBookData] = useState<any>({});
+    const [authorData, setAuthorData] = useState<any>({});
```

Comment:
Instead of any - can we type this ? The type doesn't have to exhaustive, just the basics for now.